functionary/train/train.py

```diff
@@ -82,9 +101,15 @@ def initialize_tokenizer(
 
     # Add special tokens
     tokenizer.pad_token = tokenizer.eos_token
-    special_tokens = {"additional_special_tokens": get_additional_tokens()}
+    prompt_template = get_default_prompt_template()
+    added_tokens = prompt_template.get_additional_tokens()
+    special_tokens = {"additional_special_tokens": added_tokens}
     num_new_tokens = tokenizer.add_special_tokens(special_tokens)
 
+    # add chat_template for tokenizer
+    tokenizer.chat_template = prompt_template.get_chat_template()
```

Comment:
Add chat_template to tokenizer so it is saved when we save tokenizer

Reply:
I think you can just add `tokenizer.save_pretrained(model_dir)` to the end of training to save the updated tokenizer.

Reply:
@jeffreymeetkai I think we can add the code for saving tokenizer to all checkpoints and output_dir?